lib/git.js

```diff
@@ -10,12 +10,15 @@ const debug = require('debug')('semantic-release:git');
  * @return {Array<String>} Array of modified files path.
  */
 async function filterModifiedFiles(files, execaOpts) {
-  return files.length > 0
-    ? (await execa('git', ['ls-files', '-m', '-o', ...files], execaOpts)).stdout
-        .split('\n')
-        .map(file => file.trim())
-        .filter(file => Boolean(file))
-    : [];
+  const deletedFiles = (await execa('git', ['ls-files', '-d'], execaOpts)).stdout;
```

Comment:
@pvdlg I think it would be helpful to a complementary comment, why this is done this way. What do your think?
I don't know the history of this project, but either the related issues #62 and #155 could be mentioned for reference or a short  reasoning, like "some versions of git have issues with `ls-files` so we  need to glob by ourselves and and backport deleted files, cause glob is not aware of those" 🤔